src/AbstractFixer.php

```diff
@@ -80,6 +95,44 @@ public function supports(\SplFileInfo $file)
         return true;
     }
 
+    public function configure(array $configuration = null)
+    {
+        if (!$this instanceof ConfigurationDefinitionFixerInterface) {
+            throw new \LogicException('Cannot configure using Abstact parent, child not implementing "PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface".');
+        }
+
+        if (null === $configuration) {
+            @trigger_error(
+                'Passing NULL to set default configuration is deprecated and will not be supported in 3.0, use an empty array instead.',
+                E_USER_DEPRECATED
+            );
+
+            $configuration = array();
+        }
+
+        if (null === $this->configurationDefinition || $this instanceof WhitespacesAwareFixerInterface) {
+            $this->configurationDefinition = $this->getConfigurationDefinition();
```

Comment:
with this approach one will still call `getConfigurationDefitinion()` and create new object all the time.
it is getter that should provide cached object.